src/github.com/getlantern/flashlight/http/http.go

```diff
@@ -1,125 +1,59 @@
 package http
 
 import (
-	"encoding/json"
 	"fmt"
+	"net/http"
+	"path"
+	"runtime"
+	"time"
+
 	"github.com/getlantern/flashlight/config"
 	"github.com/getlantern/flashlight/util"
 	"github.com/getlantern/golog"
 	"github.com/getlantern/proxiedsites"
-	"github.com/skratchdot/open-golang/open"
-	"net/http"
-	"reflect"
-	"time"
-
 	"github.com/getlantern/tarfs"
+	"github.com/gorilla/websocket"
+	"github.com/skratchdot/open-golang/open"
 )
 
 const (
-	UIUrl = "http://%s"
-	UIDir = "src/github.com/getlantern/ui/app"
+	UIUrl          = "http://%s"
+	LocalUIDir     = "../../../ui/app"
+	MaxMessageSize = 1024
 )
 
 var (
-	log = golog.LoggerFor("http")
+	log      = golog.LoggerFor("http")
+	upgrader = &websocket.Upgrader{ReadBufferSize: 1024, WriteBufferSize: MaxMessageSize}
+	UIDir    string
 )
 
-type JsonResponse struct {
-	Error        string   `json:"Error, omitempty"`
-	ProxiedSites []string `json:"ProxiedSites, omitempty"`
-	Global       []string `json:"Global, omitempty"`
-}
-
-type ProxiedSitesHandler struct {
+type UIServer struct {
+	Conn             *websocket.Conn
```

Comment:
We could either keep track of a pool of connections or close the former whenever a new connection is open.